src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -857,10 +871,13 @@ get_aie_partition(const axlf* top)
     pdiobj.pdi.resize(aiepdip->pdi_image.size);
     memcpy(pdiobj.pdi.data(), topbase + aiepdip->pdi_image.offset, pdiobj.pdi.size());
     for (uint32_t j = 0; j < aiepdip->cdo_groups.size; j++) {
+      std::vector<uint64_t> dpu_kernel_ids;
       auto cdop = reinterpret_cast<const cdo_group*>(topbase + aiepdip->cdo_groups.offset + j * sizeof(cdo_group));
+      auto kernel_idp = reinterpret_cast<const uint64_t*>(topbase + cdop->dpu_kernel_ids.offset);
```

Comment:
Being that this points to an array of 64-bit values, it might be easier to work with the array instead of pointers.  
It will make reading line 878 easier to read.  Something like:
`dpu_kernel_ids.push_back(kernel_idp[k]);`

Reply:
done